src/java.base/share/classes/java/util/ImmutableCollections.java

```diff
@@ -1317,6 +1317,11 @@ public int size() {
                 public Iterator<Map.Entry<K,V>> iterator() {
                     return new MapNIterator();
                 }
+
+                @Override
+                public int hashCode() {
+                    return MapN.this.hashCode();
```

Comment:
The hash code for a `Set` is defined as the sum of the elements in the `Set` (hash(`null`) == 0). The `Map. Entry` hash code is defined the same way `MapN.this.hashCode` operates so this seems right. It would be nice with a couple of tests that assert this invariant.

Reply:
Good point, this is currently missing from MOAT in Collections.

Though it can be as simple as one line here:
https://github.com/openjdk/jdk/blob/891d5aedf12e837c9a9c7cb800fb3affa7430f00/test/jdk/java/util/Collection/MOAT.java#L1329
```java
check(m.hashCode() == m.entrySet().hashCode());
```